### PR TITLE
Add FileTransfer UI and icon links license.

### PR DESCRIPTION
**Pull request by poormusic2001**

### CODE REVIEW

File: Monal/opensource.html [hunk 1]
```diff
@@ -185,6 +185,41 @@
...
+        IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+        FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+        AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+        LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+        OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+        SOFTWARE.<br><br>
+        
+        <br><br>
+        
+        <hr>
```
Comment (FriedrichAltheide) [inline Monal/opensource.html:214]: Please add a link to the main page I icons8.com above the used icon link section

File: Monal/Classes/MLFiletransfer.m [hunk 2]
```diff
@@ -604,4 +604,12 @@ +(void) markAsComplete:(id) obj
         [[NSNotificationCenter defaultCenter] postNotificationName:kMonalFiletransfersIdle object:self];
 }
 
++(BOOL) checkFileRequest:(NSNumber*) historyId
```
Comment (tmolitor-stud-tu) [inline Monal/Classes/MLFiletransfer.m:607]: This name does not really reflect what this method is doing.
I would suggest something like `isFileforHistoryIdInTransfer` or `isFileforHistoryIDInTransfer`

**Review by FriedrichAltheide — CHANGES_REQUESTED** [review]

File: Monal/Classes/MonalAppDelegate.m [hunk 3]
```diff
@@ -176,6 +178,12 @@ -(BOOL) application:(UIApplication*) application willFinishLaunchingWithOptions:
         [MLProcessLock waitForRemoteTermination:@"NotificationServiceExtension"];
     }
     
+    AVAudioSession *audioSession = AVAudioSession.sharedInstance;
```
Comment (tmolitor-stud-tu) [inline Monal/Classes/MonalAppDelegate.m:181]: Why is this here?
Shouldn't the microphone access be requested only when users try to record some audio?

File: Monal/Classes/chatViewController.m [hunk 4]
```diff
@@ -159,13 +159,39 @@ -(void) viewDidLoad
...
     self.tableviewBottom.constant += 20;
-    
-    //UTI @"public.data" for everything
-    NSString *images = (NSString *)kUTTypeImage;
-    self.imagePicker = [[UIDocumentPickerViewController alloc] initWithDocumentTypes:@[images] inMode:UIDocumentPickerModeImport];
-    self.imagePicker.allowsMultipleSelection = NO;
-    self.imagePicker.delegate = self;
 #endif
+    //UTI @"public.data" for everything
+    NSString *images = @"public.image";
```
Comment (FriedrichAltheide) [inline Monal/Classes/chatViewController.m:164]: - should be const
- Please move them to MLConstants.h

File: Monal/Classes/chatViewController.m [hunk 5]
```diff
@@ -1667,51 +1700,91 @@ -(UITableViewCell*) tableView:(UITableView*) tableView cellForRowAtIndexPath:(NS
...
+                    }
+                    else if([fileType hasPrefix:@"video/"])
+                    {
+                        fileTransferCell.transferStatus = transferVideoTypeNeedDowndload;
+                    }
+                    else if([fileType hasPrefix:@"audio/"])
+                    {
+                        fileTransferCell.transferStatus = transferAudioTypeNeedDowndload;
+                    }
+                    else if([fileType hasPrefix:@"text/"])
```
Comment (tmolitor-stud-tu) [inline Monal/Classes/chatViewController.m:1735]: Don't we need a general `else` case setting `fileTransferCell.transferStatus = transferFileTypeNeedDowndload;`? The case for application and text can be summarized into this general case as well.

File: Monal/Classes/chatViewController.m [hunk 6]
```diff
@@ -2113,6 +2211,96 @@ -(UISwipeActionsConfiguration*) tableView:(UITableView*) tableView trailingSwipe
...
+    {                
+        MLFileTransferVideoCell* videoCell = (MLFileTransferVideoCell *) [self messageTableCellWithIdentifier:@"fileTransferVideo" andInbound:inDirection fromTable:tableView];
+        NSString *videoStr = info[@"cacheFile"];
+        NSString *videoFileName = info[@"filename"];
+        [videoCell avplayerConfigWithUrlStr:videoStr fileName:videoFileName andVC:self];
+                
+        cell = videoCell;
+    }
+    else if ([info[@"mimeType"] hasPrefix:@"text/"]
+             || [info[@"mimeType"] isEqual:@"application/pdf"]//.pdf
```
Comment (FriedrichAltheide) [inline Monal/Classes/chatViewController.m:2240]: Please reuse the const names from MLConstants.h

File: Monal/Classes/chatViewController.m [hunk 7]
```diff
@@ -2113,6 +2211,96 @@ -(UISwipeActionsConfiguration*) tableView:(UITableView*) tableView trailingSwipe
...
+    else if ([info[@"mimeType"] hasPrefix:@"video/"])
+    {                
+        MLFileTransferVideoCell* videoCell = (MLFileTransferVideoCell *) [self messageTableCellWithIdentifier:@"fileTransferVideo" andInbound:inDirection fromTable:tableView];
+        NSString *videoStr = info[@"cacheFile"];
+        NSString *videoFileName = info[@"filename"];
+        [videoCell avplayerConfigWithUrlStr:videoStr fileName:videoFileName andVC:self];
+                
+        cell = videoCell;
+    }
+    else if ([info[@"mimeType"] hasPrefix:@"text/"]
```
Comment (tmolitor-stud-tu) [inline Monal/Classes/chatViewController.m:2239]: Filetransfers and links are distinct xmpp "types". All filetransfers should always have a download button (e.g. download cell).

That means you don't need this lengthy check for supported types but a general `else` case supporting all file types that have no special handling (like video, audio or image filetransfers do).

File: Monal/localization/Base.lproj/Main.storyboard [hunk 8]
```diff
@@ -1855,12 +2233,29 @@
...
+        <systemColor name="darkTextColor">
+            <color white="0.0" alpha="1" colorSpace="custom" customColorSpace="genericGamma22GrayColorSpace"/>
+        </systemColor>
         <systemColor name="darkTextColor">
             <color white="0.0" alpha="1" colorSpace="custom" customColorSpace="genericGamma22GrayColorSpace"/>
         </systemColor>
         <systemColor name="groupTableViewBackgroundColor">
             <color red="0.94901960784313721" green="0.94901960784313721" blue="0.96862745098039216" alpha="1" colorSpace="custom" customColorSpace="sRGB"/>
         </systemColor>
+        <systemColor name="groupTableViewBackgroundColor">
```
[the remark this answers is not in the document]
Comment (poormusic2001) [reply Monal/localization/Base.lproj/Main.storyboard:2250]: I use Xcode 12.2 beta 4, the newest version is 12.4. Do you use the Xcode 12.4 to develop?

File: Monal/Classes/MLFiletransfer.m (hunk 2 above)
Comment (poormusic2001) [reply Monal/Classes/MLFiletransfer.m:607]: I will modify it later.

File: Monal/opensource.html (hunk 1 above)
Comment (poormusic2001) [reply Monal/opensource.html:214]: Can you tell me... where is it?

Comment (tmolitor-stud-tu) [reply Monal/opensource.html:214]: just add `Icons from <a href="https://icons8.com/">Icons8</a>:` on the top

File: Monal/localization/Base.lproj/Main.storyboard (hunk 8 above)
[the remark this answers is not in the document]
Comment (tmolitor-stud-tu) [reply Monal/localization/Base.lproj/Main.storyboard:2250]: yes we do (that version does not add those spurious color stuff to storyboards anymore)

[the remark this answers is not in the document]
Comment (poormusic2001) [reply Monal/localization/Base.lproj/Main.storyboard:2250]: Got it.

File: Monal/opensource.html (hunk 1 above)
Comment (poormusic2001) [reply Monal/opensource.html:214]: ok

File: Monal/Classes/chatViewController.m (hunk 6 above)
Comment (poormusic2001) [reply Monal/Classes/chatViewController.m:2240]: ok

File: Monal/Classes/MonalAppDelegate.m (hunk 3 above)
Comment (poormusic2001) [reply Monal/Classes/MonalAppDelegate.m:181]: I misuse to AVPlayer. I will remove it.

**Comment by tmolitor-stud-tu** [comment]
Please create a new pull request over at https://github.com/monal-im/Monal once you resolved these issues.
We are currently in the process of merging my repo and upstream.

**Comment by poormusic2001** [comment]
> Please create a new pull request over at https://github.com/monal-im/Monal once you resolved these issues.
> We are currently in the process of merging my repo and upstream.

ok

**Comment by tmolitor-stud-tu** [comment]
We solved the review comments ourselves and merged your pull request into https://github.com/monal-im/Monal

**Pull request closed without merging**